snapcraft/cli/parts.py

```diff
@@ -68,3 +76,178 @@ def search(ctx, query):
 
     """
     remote_parts.search(" ".join(query))
+
+
+# Implemented as a generator since loading up the state could be heavy
+def _part_states_for_step(step, parts_config):
+    for part in parts_config.all_parts:
+        state = part.get_state(step)
+        if state:
+            yield (part.name, state)
+
+
+@partscli.command()
+@click.option(
+    "--provides",
+    metavar="<path>",
+    help=(
+        "Show the part that provided <path>. <path> can be relative or absolute, and "
+        "must be pointing within the staging or priming area."
+    ),
+)
+@click.option(
+    "--latest-step", is_flag=True, help="Show the most recently-completed step"
+)
+@click.option("--json", "use_json", is_flag=True, help="Print the result in json")
+def inspect(*, provides: str, latest_step: bool, use_json: bool, **kwargs):
+    """Inspect the current state of the project.
+
+    By default, this command will print a tabulated summary of the current state of the
+    project.
+    """
+    count = 0
+    for option in (provides, latest_step):
+        if option:
+            count += 1
+
+    if count > 1:
```

Comment:
I wonder if it is worth expanding click like in this last comment on this issue https://github.com/pallets/click/issues/257
Totally not a requirement for this PR, just a thought.

Reply:
I think this is simple enough not to warrant more code, but if we find we need this functionality elsewhere, it may be worth doing.